draft-thomson-http-oblivious.md

```diff
@@ -210,56 +210,69 @@ Responses are bound to responses and so consist only of AEAD-protected content.
 {{response}} describes the process for constructing and processing an
 Encapsulated Response.
 
+~~~
+Encapsulated Response {
+  Nonce (Nk),
+  AEAD-Protected Response (..),
+}
+~~~
+{: #fig-enc-response title="Encapsulated Response"}
+
+The size of the Nonce field in an Encapsulated Response corresponds to the
+size of an AEAD key for the corresponding HPKE ciphersuite.
+
 ## HPKE Encapsulation of Requests {#request}
 
 Clients encapsulate a request `request` with an HPKE public key `pkR`,
-whose Key Identifier is `keyID` as follows:
+whose wire-encoded Key Identifier is `keyID` as follows:
 
 1. Compute an HPKE context using `pkR`, yielding `context` and encapsulation
    key `enc`.
 
 2. Construct additional associated data, `aad`, by prepending a single byte
-   with a value of 0x01 to the key identifier. The key identifier length is not
+   with a value of 0x01 to the key identifier. The key identifier length is
    included in the AAD.
 
 3. Encrypt (seal) `request` with `keyID` as associated data using `context`,
    yielding ciphertext `ct`.
 
 4. Concatenate the length of `keyID` as a variable-length integer, `keyID`,
-   `enc` and `ct`, yielding an Encapsulated Request `enc_request`.
+   `enc` and `ct`, yielding an Encapsulated Request `enc_request`. Note that 
+   `enc` is of fixed-length, so there is no ambiguity in parsing `enc` and 
+   `ct`.
 
 In pseudocode, this procedure is as follows:
 
 ~~~
 enc, context = SetupBaseS(pkR, "request")
-aad = concat(0x01, keyID)
-ct = context.Seal(aad, request)
-enc_request = concat(vencode(len(keyID)), keyID, enc, ct)
+ct = context.Seal(keyID, request)
```

Comment:
I'm OK with this, but it loses the length-encoding thing (and the lines don't match the numbered list any more).